class.jetpack-xmlrpc-server.php

```diff
@@ -70,9 +70,31 @@ function xmlrpc_methods( $core_methods ) {
 	function bootstrap_xmlrpc_methods() {
 		return array(
 			'jetpack.verifyRegistration' => array( $this, 'verify_registration' ),
+			'jetpack.remoteAuthorize' => array( $this, 'remote_authorize' ),
 		);
 	}
 
+	function remote_authorize( $request ) {
```

Comment:
My latest commits, here on on Phab D972, should take care of tieing the user to the secrets.